Source/Surfaces/PrincipalCurvature.f90

```diff
@@ -0,0 +1,267 @@
+!
+!////////////////////////////////////////////////////////////////////////
+!
+!      PrincipalCurvature.f90
+!      Created: March 19, 2025 at 10:54 AM 
+!      By: David Kopriva  
+!
+!////////////////////////////////////////////////////////////////////////
+!
+   Module PrincipalCurvatureModule
+   USE SMConstants
+   IMPLICIT NONE  
+! 
+!-----------------------------------------------------------------------
+! Two procedures are included with which to compute the maximum 
+! absolute value ofthe principal curvature at a point (x,y) 
+! of a surface z = f(x,y). 
+!
+! (1) PrincipalCurvature(gradF, HessF)
+!         REAL(KIND=RP) :: gradF(2)
+!         REAL(KIND=RP) :: hessF(2,2)
+!
+!    The first takes the gradient and Hessian at the point (x,y) and
+!    returns the result. Use this if the gradient and hessian can
```

Comment:
```suggestion
!    returns the result. Use this if the gradient and Hessian can
```

Reply:
I'll make this change in the next commit.